clearnode/app_session_service.go

```diff
@@ -429,6 +433,10 @@ func (s *AppSessionService) CreateAppSession(params *CreateAppSessionParams, rpc
 				return err
 			}
 
+			if err := ensureWalletHasZeroChannelAllocation(tx, walletAddress); err != nil {
+				return err
+			}
```

Comment:
Currently the check resides inside the "allocations loop", meaning if a participant does NOT have an allocation on channel creation, they may be included in the app session successfully even if they have a non-empty allocation channel.

However, this is mitigated by the same restriction inside the "deposit" app state submission, ensuring a user with non-empty channel can NOT allocate their funds elsewhere.

Still, this may be worth our attention as current logic introduces inconsistency, because a user with a non-empty channel can join an app session.